### ISSUE TITLE
replace map use with None as first parameter on `twisted.mail` package

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5006]
|<img alt="nueces's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| nueces reported|
|-|-|
|Trac ID|trac#5006|
|Type|enhancement|
|Created|2011-03-27 20:53:05Z|
|Branch|https://github.com/twisted/twisted/tree/replace-map-None-5006|

The use of map with None as first parameter don't have support on python3. So I'm rewriting it to make more easy the upgrade to python3.

Attachments:

* [map-mail.path](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5de/5de34098fb80dc620a0633997e1a9c362d5ee468/41882d167e7ad682166beddb555edbf55d56131a.path) (415 bytes) - added by nueces on 2011-03-27 20:53:46Z - path

<details><summary>Searchable metadata</summary>

```
trac-id__5006 5006
type__enhancement enhancement
reporter__nueces nueces
priority__normal normal
milestone__ 
branch__branches_replace_map_None_5006 branches/replace-map-None-5006
branch_author__mithrandi mithrandi
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__py3k py3k
time__1301259185000000 1301259185000000
changetime__1301769852000000 1301769852000000
version__None None
owner__mithrandi mithrandi
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Is this a duplicate of [#4555](https://github.com/twisted/twisted/issues/4555)?  I like the reduced scope, but let's try to avoid re-doing work that's been done already.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_4"></a>|@mithrandi commented|
|-|-|

I don't believe this is a duplicate of [[#4555](https://github.com/twisted/twisted/issues/4555)](https://github.com/twisted/twisted/issues/4555). map with None as the first parameter is similar to zip(), except with different behaviour when the sequences are of differing lengths. #4555 is about uses of map that expect the side effects of the map function to be executed immediately, instead of lazily.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

> [[[#4555](https://github.com/twisted/twisted/issues/4555)](https://github.com/twisted/twisted/issues/4555)](https://github.com/twisted/twisted/issues/4555) is about uses of map that expect the side effects of the map function to be executed immediately, instead of lazily.

The patch attached to #4555 contains exactly the same change as the patch attached to this ticket, though.  So I think _someone_ thought #4555 covered the change this ticket is about. :)

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_7"></a>|@mithrandi commented|
|-|-|

(In [31514]) Branching to 'replace-map-None-5006'

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_8"></a>|@mithrandi set owner to @mithrandi|
|-|-|

Okay, so it seems like the consensus is that applying these patches in smaller chunks is better than the one-huge-patch over at [#4555](https://github.com/twisted/twisted/issues/4555). This patch chunk looks fine by itself, and the affected code seems to be covered by the tests already; so I'm going to go ahead and apply this.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_9"></a>|@mithrandi set status to `closed`|
|-|-|

(In [31518]) Merge replace-map-None-5006.

Author: nueces
Reviewer: mithrandi
Fixes: [#5006](https://github.com/twisted/twisted/issues/5006)

Replace use of map(None, ...) in twisted.mail.smtp.
